packages/data-point/README.md

```diff
@@ -3195,6 +3125,85 @@ app.listen(3000, function () {
 })
 ```
 
+## <a name="patterns-best-practices">Patterns and Best Practices</a>
+
+This section documents some of the patterns and best practices we have found useful while using DataPoint.
+
+### Parameterize a FunctionReducer
+
+You may use a [higher order function](https://medium.com/javascript-scene/higher-order-functions-composing-software-5365cf2cbe99) to parameterize a [FunctionReducer](#function-reducer). To do this you will create a function that **must** return a [FunctionReducer](#function-reducer).
+
+```js
+// sync
+const name = (param1, param2, ...) => (acc:Accumulator) => {
+  return newValue
+}
+// async via promise
+const name = (param1, param2, ...) => (acc:Accumulator) => {
+  return Promise.resolve(newValue)
+}
+// async via callback
+const name = (param1, param2, ...) => (acc:Accumulator, next:function) => {
+  next(error:Error, newValue:*)
+}
+```
+
+<details>
+  <summary>Higher Order Reducer Example</summary>
+  
+  ```js
+  const addStr = (value) => (acc) => {
+    return acc.value + value
+  }
+  
+  dataPoint
+    .transform(addStr(' World!!'), 'Hello')
+    .then((acc) => {
+      assert.equal(acc.value, 'Hello World!!')
+    })
+  ```
+</details>
+
+
+Example at: [examples/reducer-function-closure.js](examples/reducer-function-closure.js)
+
+### Keeping your FunctionReducers pure
+
+When it comes to creating your own reducer functions you want them to be [pure functions](https://medium.com/javascript-scene/master-the-javascript-interview-what-is-a-pure-function-d1c076bec976#.r4iqvt9f0); this is so they do not produce any side effects. 
+
+In the context of a [FunctionReducer](#function-reducer) this means, you should never change directly (or indirectly) the value of the [Accumulator.value](#accumulator). 
```

Comment:
The comma after `this means` is not necessary

Reply:
@paulmolluzzo  thanks! reading it for a second time...  thoughts on: 

In the context of a [FunctionReducer](#function-reducer) it means you should never change directly (or indirectly) the value of the [Accumulator.value](#accumulator).

Reply:
Sure, that reads just as well. 👍